framework/CodeInterfaces/SCALE/ScaleInterface.py

```diff
@@ -0,0 +1,182 @@
+      # Check if the input requested by the sequence has been found
+      if self.sequence.count('triton') != len(inputDict['triton']):
+        raise IOError('triton input file has not been found. Files type must be set to "triton"!')
+      if self.sequence.count('origen') != len(inputDict['origen']):
+        raise IOError('origen input file has not been found. Files type must be set to "origen"!')
+      return inputDict
+
+  def generateCommand(self, inputFiles, executable, clargs=None, fargs=None):
+    """
+      Generate a command to run Mammoth using an input with sampled variables
+      See base class.  Collects all the clargs and the executable to produce the command-line call.
+      Returns tuple of commands and base file name for run.
+      Commands are a list of tuples, indicating parallel/serial and the execution command to use.
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have
+        been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path (e.g. /home/path_to_executable/code.exe)
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input
+        (e.g. under the node < Code >< clargstype = 0 input0arg = 0 i0extension = 0 .inp0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify
+        in the input (e.g. under the node < Code >< fargstype = 0 input0arg = 0 aux0extension = 0 .aux0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the
+        code (string), returnCommand[1] is the name of the output root
+    """
+    inputDict = self.findInps(inputFiles)
+    executeCommand = []
+    for seq in self.sequence:
+      self.outputRoot[seq.lower()] = inputDict[seq.lower()][0].getBase()
+      executeCommand.append(('parallel',executable+' '+inputDict[seq.lower()][0].getFilename()))
+    returnCommand = executeCommand, self.outputRoot.values()[-1]
+    return returnCommand
+
+  def createNewInput(self, currentInputFiles, origInputFiles, samplerType, **Kwargs):
+    """
+      Generates new perturbed input files for Scale sequences
+      @ In, currentInputFiles, list,  list of current input files
+      @ In, origInputFiles, list, list of the original input files
+      @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
+      @ In, Kwargs, dict, dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
+        where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
+      @ Out, newInputFiles, list, list of new input files (modified or not)
+    """
+    if 'dynamiceventtree' in str(samplerType).lower():
```

Comment:
dynamicEventTree

Reply:
it  is a check on the sampler type...It is not a variable. I use this approach because we have
DynamicEventTree, HybridDynamicEventTree etc.